tds.go

```diff
@@ -1011,10 +1011,10 @@ initiate_connection:
 				success = true
 				sess.loginAck = token
 			case error:
-				return nil, fmt.Errorf("Login error: %s", token.Error())
+				return nil, token
 			case doneStruct:
 				if token.isError() {
-					return nil, fmt.Errorf("Login error: %s", token.getError())
```

Comment:
we could potentially use wrapping here : 
`return nil, fmt.Errorf("Login error: %w", token.getError())`
that would keep the same current format (add the error string), but also provide the full underlying error when using errors.Is/As(...)